homeassistant/components/yolink/binary_sensor.py

```diff
@@ -0,0 +1,98 @@
+"""YoLink BinarySensor."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from yolink.device import YoLinkDevice
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_COORDINATOR, ATTR_DEVICE_DOOR_SENSOR, DOMAIN
+from .coordinator import YoLinkCoordinator
+from .entity import YoLinkEntity
+
+
+@dataclass
+class YoLinkBinarySensorEntityDescriptionMixin:
+    """Mixin for device type."""
+
+    exists_fn: Callable[[YoLinkDevice], bool] = lambda _: True
+
+
+@dataclass
+class YoLinkBinarySensorEntityDescription(
+    YoLinkBinarySensorEntityDescriptionMixin, BinarySensorEntityDescription
+):
+    """YoLink BinarySensorEntityDescription."""
+
+    value: Callable = lambda state: state
```

Comment:
Is this really a good default? Maybe we should move this attribute to the mixin and make it required.
```suggestion
    value: Callable[[str], bool] = lambda state: bool(state)
```